src/System.ComponentModel.TypeConverter/src/System/Drawing/ImageFormatConverter.cs

```diff
@@ -0,0 +1,155 @@
+                else if (strFormat.Equals(ImageFormat.Gif.ToString(), StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Gif;
+                else if (strFormat.Equals(ImageFormat.Icon.ToString(), StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Icon;
+                else if (strFormat.Equals(ImageFormat.Jpeg.ToString(), StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Jpeg;
+                else if (strFormat.Equals(ImageFormat.MemoryBmp.ToString(), StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.MemoryBmp;
+                else if (strFormat.Equals(ImageFormat.Png.ToString(), StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Png;
+                else if (strFormat.Equals(ImageFormat.Tiff.ToString(), StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Tiff;
+                else if (strFormat.Equals(ImageFormat.Wmf.ToString(), StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Wmf;
+            }
+            else
+            {
+                // case #3, this is probably a short format
+                if (strFormat.Equals("Bmp", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Bmp;
+                else if (strFormat.Equals("Emf", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Emf;
+                else if (strFormat.Equals("Exif", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Exif;
+                else if (strFormat.Equals("Gif", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Gif;
+                else if (strFormat.Equals("Icon", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Icon;
+                else if (strFormat.Equals("Jpeg", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Jpeg;
+                else if (strFormat.Equals("MemoryBmp", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.MemoryBmp;
+                else if (strFormat.Equals("Png", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Png;
+                else if (strFormat.Equals("Tiff", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Tiff;
+                else if (strFormat.Equals("Wmf", StringComparison.OrdinalIgnoreCase))
+                    return ImageFormat.Wmf;
+            }
+            // last case, this is an unknown string
+            return base.ConvertFrom(context, culture, value);
+        }
+
+        public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
+        {
+            if (value is ImageFormat imgFormat)
+            {
+                string strFormat = null;
+                if (imgFormat.Guid.Equals(ImageFormat.Bmp.Guid))
```

Comment:
ImageFormat.ToString() already does this comparisons. Delete this and in the if == typeof(string) just return imgFormat.ToString();

https://github.com/dotnet/corefx/blob/master/src/System.Drawing.Common/src/System/Drawing/Imaging/ImageFormat.cs#L162